### ISSUE TITLE
timing issues when saving notes

### ISSUE THREAD

**Issue by boz** [laurengordonfahn/simple_notebook_project#6]
When [creating a new note](https://github.com/laurengordonfahn/simple_notebook_project/blob/master/helper_functions.py#L19-L25), the object is inserted into the database and then the newest note is returned.

Imagine if two requests came in at the same time to create a note.  The database operations might end up looking like:

1. insert note
1. insert note
1. find newest note
1. find newest note

In other words, both requests requests would return the same note.

I believe that the `Note` object which was used in `db.session.add(note)` will have fully populated `id`, `created_at`, `updated_at` fields after `db.session.commit()`, which means it can just be returned.  If this is not the case then find another method which does not suffer from the timing issues the current implementation has.

**Comment by laurengordonfahn** [comment]
I have successfully dealt with all the issues you have raised but one. I will need to do some further reading on how to use pull request with the issue number to close my issues 'not by hand' I shall look into this tomorrow! This was great, and expanding. Thank you
